bindings/elixir/native/src/ckzg_wrap.c

```diff
@@ -0,0 +1,698 @@
+    if (KZGSETTINGS_RES_TYPE == NULL) return -1;
+
+    ckzg_atoms.ok = enif_make_atom(env, "ok");
+    ckzg_atoms.error = enif_make_atom(env, "error");
+    ckzg_atoms.a_true = enif_make_atom(env, "true");
+    ckzg_atoms.a_false = enif_make_atom(env, "false");
+    ckzg_atoms.kzg_badargs = enif_make_atom(env, "kzg_badargs");
+    ckzg_atoms.kzg_error = enif_make_atom(env, "kzg_error");
+    ckzg_atoms.kzg_malloc = enif_make_atom(env, "kzg_malloc");
+    ckzg_atoms.kzg_unknown_error = enif_make_atom(env, "kzg_unknown_error");
+    ckzg_atoms.incorrect_arg_count = enif_make_atom(env, "incorrect_arg_count");
+    ckzg_atoms.invalid_precompute_arg = enif_make_atom(env, "invalid_precompute_arg");
+    ckzg_atoms.bad_file_string_length = enif_make_atom(env, "bad_file_string_length");
+    ckzg_atoms.out_of_memory = enif_make_atom(env, "out_of_memory");
+    ckzg_atoms.bad_file_string_arg = enif_make_atom(env, "bad_file_string_arg");
+    ckzg_atoms.failed_to_open_file = enif_make_atom(env, "failed_to_open_file");
+    ckzg_atoms.failed_alloc_settings_resource = enif_make_atom(
+        env, "failed_alloc_settings_resource"
+    );
+    ckzg_atoms.blob_not_binary = enif_make_atom(env, "blob_not_binary");
+    ckzg_atoms.invalid_blob_length = enif_make_atom(env, "invalid_blob_length");
+    ckzg_atoms.z_not_binary = enif_make_atom(env, "z_not_binary");
+    ckzg_atoms.invalid_z_length = enif_make_atom(env, "invalid_z_length");
+    ckzg_atoms.y_not_binary = enif_make_atom(env, "y_not_binary");
+    ckzg_atoms.invalid_y_length = enif_make_atom(env, "invalid_y_length");
+    ckzg_atoms.commitment_not_binary = enif_make_atom(env, "commitment_not_binary");
+    ckzg_atoms.invalid_commitment_length = enif_make_atom(env, "invalid_commitment_length");
+    ckzg_atoms.proof_not_binary = enif_make_atom(env, "proof_not_binary");
+    ckzg_atoms.invalid_proof_length = enif_make_atom(env, "invalid_proof_length");
+    ckzg_atoms.failed_get_settings_resource = enif_make_atom(env, "failed_get_settings_resource");
+    ckzg_atoms.expected_same_array_size = enif_make_atom(env, "expected_same_array_size");
+    ckzg_atoms.cell_indices_not_list = enif_make_atom(env, "cell_indices_not_list");
+    ckzg_atoms.cells_not_list = enif_make_atom(env, "cells_not_list");
+    ckzg_atoms.cell_indices_value_not_uint64 = enif_make_atom(env, "cell_indices_value_not_uint64");
+    ckzg_atoms.cells_value_not_binary = enif_make_atom(env, "cells_value_not_binary");
+    ckzg_atoms.invalid_cell_length = enif_make_atom(env, "invalid_cell_length");
+
+    return 0;
+}
+
+static ERL_NIF_TERM load_trusted_setup_nif(ErlNifEnv *env, int argc, const ERL_NIF_TERM argv[]) {
+    if (argc != 2) return make_error(env, ckzg_atoms.incorrect_arg_count);
+
+    unsigned long precompute;
+    if (!enif_get_ulong(env, argv[1], &precompute))
+        return make_error(env, ckzg_atoms.invalid_precompute_arg);
+
+    unsigned int len;
+    if (!enif_get_string_length(env, argv[0], &len, ERL_NIF_LATIN1))
```

Comment:
note that [`enif_get_string_length`](https://www.erlang.org/doc/apps/erts/erl_nif.html#enif_get_string_length) was only added since OTP 26.0 which was released in [May 2023](https://www.erlang.org/patches/otp-26.0), not sure if we want to support older versions too?